File: AcupayEmployee/Core Forms/JobLevel/Entity/JobLevel.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.JobLevels

    <Table("joblevel")>
    Public Class JobLevel

        <Key, Column(Order:=1)>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        Public Property Created As Date

        Public Property CreatedBy As Integer?

        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property JobCategoryID As Integer?

        Public Property Name As String

        Public Property Points As Integer

        Public Property SalaryRangeFrom As Decimal

        Public Property SalaryRangeTo As Decimal

        <ForeignKey("JobCategoryID")>
        Public Overridable Property JobCategory As JobCategory

    End Class

End Namespace

File: AccuPay.Core/Entities/JobLevel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Core.Entities
{
    [Table("joblevel")]
    public class JobLevel
    {
        [Key]
        [Column(Order = 1)]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        public int? OrganizationID { get; set; }

        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public int? JobCategoryID { get; set; }

        public string Name { get; set; }

        public int Points { get; set; }

        public decimal SalaryRangeFrom { get; set; }

        public decimal SalaryRangeTo { get; set; }

        [ForeignKey("JobCategoryID")]
        public virtual JobCategory JobCategory { get; set; }
    }
}